File: app/src/main/java/com/brainwellnessspa/Utility/MyService.java
package com.brainwellnessspa.Utility;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import androidx.annotation.Nullable;

public class MyService extends Service {
    private MyThread mythread;
    public boolean isRunning = false;

    @Override
    public void onCreate() {
        super.onCreate();
        ////Log.d("", "onCreate");
        mythread = new MyThread();
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (!isRunning) {
            mythread.start();
            isRunning = true;
        }
        return START_STICKY;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (!isRunning) {
            mythread.interrupt();
            mythread.stop();
        }
    }

    class MyThread extends Thread {
        static final long DELAY = 3000;
        @Override
        public void run() {
            while (isRunning) {
                try {
//                    YupITApplication.getToken();
                    Thread.sleep(DELAY);
                } catch (InterruptedException e) {
                    isRunning = false;
                    e.printStackTrace();
                }
            }
        }
    }
}

File: app/src/main/java/com/brainwellnessspa/utility/MyService.kt
package com.brainwellnessspa.utility

import android.app.Service
import android.content.Intent
import android.os.IBinder

class MyService : Service() {
    private var mythread: MyThread? = null
    var isRunning = false
    val delay: Long = 3000
    override fun onCreate() {
        super.onCreate() ////Log.d("", "onCreate");
        mythread = MyThread()
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        if (!isRunning) {
            mythread!!.start()
            isRunning = true
        }
        return START_STICKY
    }

    override fun onDestroy() {
        super.onDestroy()
        if (!isRunning) {
            mythread!!.interrupt()
            mythread!!.stop()
        }
    }

    internal inner class MyThread : Thread() {
        override fun run() {
            while (isRunning) {
                try { //                    YupITApplication.getToken();
                    sleep(delay)
                } catch (e: InterruptedException) {
                    isRunning = false
                    e.printStackTrace()
                }
            }
        }
    }
}